src/nvim/api/keysets_defs.h

```diff
@@ -361,12 +382,13 @@ typedef struct {
 
 typedef struct {
   OptionalKeys is_set__xdl_diff_;
-  LuaRef on_hunk;
+  LuaRefOf((Integer start_a, Integer count_a, Integer start_b, Integer count_b),
```

Comment:
non-blocking, just for your consideration:
```suggestion
  LuaFn((Integer start_a, Integer count_a, Integer start_b, Integer count_b),
```

Reply:
I chose `LuaRefOf` to be consistent with `DictOf` and `ArrayOf`. To me it would make more sense to rename `LuaRef` to `LuaFun` and `LuaRefOf` to `LuaFunOf`. The only difference with the `Of` variant is extra metadata.

Or does `LuaRef` mean something more general?

Reply:
> To me it would make more sense to rename `LuaRef` to `LuaFun` and `LuaRefOf` to `LuaFunOf`

That also SGTM.

> Or does `LuaRef` mean something more general?

I think it's a carry-over from Vimscript/typval where function references were called "FuncRefs".

Reply:
`LuaRef` can be any lua value. We use it for tables in some places. It is fine to have specialized typedefs on top of course.